File: PGProblemSolving/Hash/위장_42578_Lv2.swift
//
//  위장_42578_Lv2.swift
//  PGProblemSolving
//
//  Created by MinKyeongTae on 2020/03/11.
//  Copyright © 2020 MinKyeongTae. All rights reserved.
//

// MARK: - 위장 42578 Lv2
// MARK: Hash 문제풀이

/*
import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var dic = [String:Int]()
    var Ans = 1
    for i in clothes.indices {
        let type = clothes[i][1]
        dic[type] = (dic[type] ?? 0) + 1
    }
    
    if dic.count == 1 { return clothes.count }
    
    for (key, value) in dic {
        Ans *= (value + 1)
    }
    
    return Ans - 1
}
*/

File: PGProblemSolving/Hash/위장_42578_Lv2.cpp
//
//  위장.cpp
//  ProgrammersAlgorithmPractice
//
//  Created by MinKyeongTae on 10/09/2019.
//  Copyright © 2019 Min Kyeong Tae. All rights reserved.
//

// MARK: 위장 Lv2 문제풀이
#if 0
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map<string,int> MP;
    for(auto &v : clothes) MP[v[1]]++;
    for(auto &mp : MP) answer *= (mp.second+1);
    return answer-1;
}
#endif
